wweval/R/ms_fig3.R

```diff
@@ -1,3 +1,104 @@
+#' Get a summary of a single states crps
+#'
+#' @param scores tibble of crps scores by location, forecast date, model,
+#' horizon day
+#' @param locs_to_plot the locations we want summaries for
+#' @param fig_file_dir string indicating directory to save fig in
+#'
+#' @return a table with mean crps for each model and the relative crps
+get_summary_table_fig3 <- function(scores,
+                                   locs_to_plot,
+                                   fig_file_dir) {
+  scores_locs_long <- scores |>
+    dplyr::filter(
+      location %in% locs_to_plot
+    ) |>
+    dplyr::group_by(model, location) |>
+    dplyr::summarize(
+      mean_crps = mean(crps)
+    )
+  scores_locs <- scores_locs_long |>
+    tidyr::pivot_wider(
+      id_cols = c("location"),
+      names_from = "model",
+      names_prefix = "mean_crps_",
+      values_from = mean_crps
+    ) |>
+    dplyr::mutate(
+      rel_crps_means = mean_crps_ww / mean_crps_hosp
+    )
+
+  colors <- plot_components()
+  p <- ggplot(scores_locs_long) +
+    geom_bar(aes(x = model, y = mean_crps, fill = model),
+      stat = "identity",
+      position = "dodge"
+    ) +
+    scale_fill_manual(values = colors$model_colors) +
+    facet_wrap(~location) +
+    get_plot_theme(
+      x_axis_dates = TRUE,
+      y_axis_title_size = 8,
+      y_axis_text_size = 6
+    ) +
+    xlab("") +
+    ylab("Mean CRPS")
+
+  ggsave(p,
+    filename = file.path(fig_file_dir, "sfig_bar_chart_mean_crps_3_locs.png"),
+    width = 7, height = 4
+  )
+
+  raw_rel_scores <- scores |>
+    dplyr::filter(
+      location %in% locs_to_plot
+    ) |>
+    tidyr::pivot_wider(
+      id_cols = c("location", "date", "forecast_date"),
+      names_from = "model",
+      names_prefix = "crps_",
+      values_from = crps
+    ) |>
+    dplyr::mutate(rel_crps = crps_ww / crps_hosp) |>
+    dplyr::group_by(location) |>
+    dplyr::summarize(mean_rel_crps = mean(rel_crps, na.rm = TRUE))
```

Comment:
Are we using this anywhere? In general, means of relative scores should be geometric.
```suggestion
    dplyr::summarize(mean_rel_crps = mean(rel_crps, na.rm = TRUE))
```

Reply:
We aren't reporting this anymore, no. There are still summary tables that have mean relative crps but very explicitly in the text we are taking the relative of means.

Reply:
I will remove this from the function